src/Storefront/Data/Service/VaultDataService.php

```diff
@@ -61,6 +61,9 @@ public function buildData(SalesChannelContext $context): ?VaultData
         return $struct;
     }
 
+    /**
+     * @deprecated tag:v10.0.0 - Will be removed. Use `CustomerVaultTokenRoute::getVaultToken` instead
+     */
```

Comment:
now that trunk is being prepared for becoming `v10.0.0`, we can already break here, as long as this deprecation gets introduced on the `6.6.x` branch beforehand.